src/fparser/two/symbol_table.py

```diff
@@ -604,6 +633,11 @@ def lookup(self, name):
         lname = name.lower()
         if lname in self._data_symbols:
             return self._data_symbols[lname]
+        for module in self._modules.values():
+            try:
+                return module.lookup(lname)
+            except KeyError:
+                pass
```

Comment:
It's not clear to me why you test parent scope and only parent scope in the code below. Could you please explain in a comment. Is it to do with routines being inside modules?

Reply:
Oh yes, that whole method was rather comment-light. Essentially we recurse upwards (`self.parent.lookup` calls this method on the parent symbol table) - I've added a comment.